src/devices/tuya.ts

```diff
@@ -25,6 +25,266 @@ const fzZosung = zosung.fzZosung;
 const tzZosung = zosung.tzZosung;
 const ez = zosung.presetsZosung;
 
+const PJ1203A = {
+
+    //
+    // The PJ-1203A is sending quick sequence of messages containing a single datapoint.
+    // A sequence occurs every `update_frequency` seconds (10s by default)
+    //
+    // A typical sequence is composed of two identical groups for channel a and b.
+    //
+    //     102 energy_flow_a
+    //     112 voltage
+    //     113 current_a
+    //     101 power_a
+    //     110 power_factor_a
+    //     111 ac_frequency
+    //     115 power_ab
+    //     ---
+    //     104 energy_flow_b
+    //     112 voltage
+    //     114 current_b
+    //     105 power_b
+    //     121 power_factor_b
+    //     111 ac_frequency
+    //     115 power_ab
+    //
+    // It should be noted that when no current is detected on channel x then
+    // energy_flow_x is not emited and current_x==0, power_x==0 and power_factor_x==100.
+    //
+    // The other datapoints are emitted every few minutes.
+    //
+    // There is a known issue on the _TZE204_81yrt3lo (with appVersion 74, stackVersion 0 and hwVersion 1).
+    // The energy_flow datapoints are (incorrectly) emited during the next update. This is quite problematic
+    // because that means that the direction can be inverted for up to update_frequency seconds.
+    //
+    // The features implemented here are
+    //   - cache the datapoints for each channel and publish them together.
+    //   - (OPTIONAL) solve the issue described above by waiting for the next energy flow datapoint
+    //     before publishing the cached channel data.
+    //   - (OPTIONAL) provide signed power instead of energy flow.
+    //   - detect missing or reordered Zigbee message using the Tuya 'seq' attibute and invalidate
+    //     cached data accordingly.
+    //
+
+    get_private_state: (device: Zh.Device) => {
+        let priv = globalStore.getValue(device, 'private_state');
```

Comment:
You can use the third `default_unknown` argument here to return a default value instead of ` if (priv===undefined) {`

Reply:
Ok I'll do that. For information, I choose to do it that way for two reasons: First to avoid creating and discarding a new default state every time `get_private_state` is called (is there a way to customize the device at startup? I could not find how) and second because I was planning for future customization of the initial `private_state` for some versions of the device (e.g. different `sec_inc` or different ordering of the DPs).

Reply:
I see, then let's leave it as is.